### PR TITLE
Move common utility functions to a common library

### PR DESCRIPTION
**Pull request by srittau**
Prerequisite for #12772

Unfortunately, this is a bit hacky, but I don't have a better idea.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -162,7 +162,7 @@ extra-standard-library = [
     "pyexpat",
     "zoneinfo",
 ]
-known-first-party = ["_metadata", "_utils"]
+known-first-party = ["ts_utils", "_utils"]
```
Comment (JelleZijlstra) [inline pyproject.toml:165]: Can we remove `_utils`?

File: tests/mypy_test.py [hunk 2]
```diff
@@ -22,8 +22,11 @@
 import tomli
 from packaging.requirements import Requirement
 
-from _metadata import PackageDependencies, get_recursive_requirements, read_metadata
-from _utils import (
+sys.path += ["lib"]
```
Comment (JelleZijlstra) [inline tests/mypy_test.py:25]: This is pretty unfortunate, and also seems to create a dependency on the working directory from which the code is invoked. Could we instead add `pyproject.toml` to the lib directory to make it installable, then add it as a local dependency to requirements.txt?

Comment (srittau) [reply tests/mypy_test.py:25]: Another option is to use `Path(__file__).parent.parent / "lib"`. On the other hand, most scripts are already assuming that they're called from the root typeshed directory, as they are using `Path("stubs")` and similar constructs.

Comment (Avasam) [reply tests/mypy_test.py:25]: Personally I'd rather scripts be agnostics of their working directory when possible.

File: pyproject.toml (hunk 1 above)
Comment (Avasam) [reply pyproject.toml:165]: tbf I only named it `_utils` to reflect this config, and as it is right know, it wouldn't even affect the order as those scripts are only using stdlib.

I also see no issue moving or renaming `scripts/sync_protobuf/_utils.py` if that's wanted.

**Comment by srittau** [comment]
Third-party test failure is unrelated (gdb).

**Review by Avasam — APPROVED** [review]
Personally I'm happy with this

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**